Frameworks/Foundation/NSThread.mm

```diff
@@ -356,8 +356,15 @@ + (NSThread*)currentThread {
     NSThread* currentThread = [[self class] _threadObjectFromCurrentThread];
 
     if (currentThread == nil) {
-        currentThread = [NSThread new];
+        // Underlying thread created outside the NSThread APIs
+
+        // So create a new instance
+        currentThread = [[NSThread alloc] init];
         [currentThread _associateWithCurrentThread];
```

Comment:
That makes sense. I'll include that + some cleanups in the next commit - I can see there's e.g. an unused pthread tls key in that file too.